kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/HttpClientUtils.java

```diff
@@ -139,6 +139,16 @@ public Response intercept(Chain chain) throws IOException {
                 }
             }
 
+            if (config.getUserAgent() != null && !config.getUserAgent().isEmpty()) {
+                httpClient.interceptors().add(new Interceptor() {
```

Comment:
Yeah let's add it as a network interceptor please - seen it done in a few places for user agent & seems to be the accepted way. Shame it's not just a simple case of configuring http client but it works...